text/0224-experimental-semantic-conventions.md

```diff
@@ -0,0 +1,117 @@
+What should OpenTelemetry do with semantic convention proposals when there's unclear evidence that a particular signal is necessary or ubiquitous?
+
+## Explanation
+
+This proposal generally follows the recommendations of [RFC6648 section 4 “Recommendations for Protocol Designers.”](https://www.rfc-editor.org/rfc/rfc6648#section-4)
+RFC6648 recommends a registry of names be established which standardizes usage and prevents collisions.
+This OTEP separates a dictionary of semantic convention attributes from the semantic conventions, to be referenced by semantic conventions when necessary, and establishes a procedure for registering new and experimental semantic convention attributes.
+
+RFC6648 further recommends that X- prefix (or similar structure) MUST NOT be defined as experimental or nonstandard, and that names missing the X- prefix MUST NOT be defined as stable or standard.
+This is addressed below in [alternatives considered](#alternatives-considered).
+
+## Definitions
+
+**Attribute Dictionary**: The attribute dictionary contains all semantic convention attributes, including their names and value definitions.
+
+**Registered Attribute**: A registered attribute is any attribute which is in the attribute dictionary.
+
+## Proposal
+
+A semantic convention attribute dictionary will be established which is maintained in a separate file or directory from the semantic conventions, seeded with all existing semantic convention attributes.
+All semantic conventions MUST refer to the attribute dictionary for all attributes used by that semantic convention.
+Semantic conventions MUST NOT establish an attribute not in the attribute dictionary without explicit approval of the TC with the reasoning documented directly in that convention.
+A procedure will be established by the technical committee (TC), or an entity designated by the TC, for registration of new semantic convention attributes.
+Any attributes not intended strictly for private use SHOULD be registered in the attribute dictionary.
+
+Instrumentation authors SHOULD assume that any attribute they create may become standardized, public, commonly deployed, or usable across multiple OpenTelemetry clients, distributions, or receivers.
+They SHOULD choose attribute keys that are descriptive, relevant, and that they have reason to believe are not currently in use or already proposed.
+Before creating a new attribute, the attribute author SHOULD review the attribute dictionary to see if an attribute already exists which fits the intended use case.
+If such a registered attribute exists, then it SHOULD be preferred over the creation of new attributes unless there is sufficient reason to create a new attribute.
+Any new attributes SHOULD be registered in the attribute dictionary before an instrumentation using the attribute is released.
```

Comment:
The idea is to prevent collisions. If some instrumentation starts emitting `db.my_attribute` but doesn't register it, another instrumentation may emit the same attribute. I think the blanket normative language is fine. Carving exemptions into the language for things we don't control seems unnecessary to me.